src/pip/_internal/vcs/__init__.py

```diff
@@ -228,14 +228,15 @@ def get_url_rev_and_auth(self, url):
 
         Returns: (url, rev, (username, password)).
         """
-        error_message = (
-            "Sorry, '%s' is a malformed VCS url. "
-            "The format is <vcs>+<protocol>://<url>, "
-            "e.g. svn+http://myrepo/svn/MyApp#egg=MyApp"
-        )
-        assert '+' in url, error_message % url
-        url = url.split('+', 1)[1]
         scheme, netloc, path, query, frag = urllib_parse.urlsplit(url)
+        if '+' not in scheme:
+            raise ValueError(
+                "Sorry, '{}' is a malformed VCS url. "
```

Comment:
Minor nit: you could use `"Sorry, {!r} is a malformed VCS url. "`. This is almost the same thing, except that if the URL contains a `'` character, the URL will be quoted with `"` rather than `'`, and that if the user somehow passed binary data as the URL (think: misbehaving script) it would not stuff up their terminal.

Reply:
Thanks. Yes, I agree. But I was trying to limit myself to fixing just the one issue I noticed to keep the PR more focused. For example, I also think the message itself could be made better and more concise, but that would trigger another discussion.